packages/block-library/src/button/edit.js

```diff
@@ -133,10 +134,60 @@ function ButtonEdit( props ) {
 
 	const [ isEditingURL, setIsEditingURL ] = useState( false );
 	const isURLSet = !! url;
-	const opensInNewTab = linkTarget === NEW_TAB_TARGET;
-	const nofollow = !! rel?.includes( NOFOLLOW_REL );
+
 	const isLinkTag = 'a' === TagName;
 
+	// Defines how block attributes map to link value and vice versa.
+	const linkValueAttrsToDataMapping = {
+		url: 'url',
+		opensInNewTab: {
+			dataKey: 'linkTarget',
+			toLink: ( value ) => value === NEW_TAB_TARGET,
+			toData: ( value ) => ( value ? NEW_TAB_TARGET : undefined ),
+		},
+		nofollow: {
+			dataKey: 'rel',
+			toLink: ( value ) => value?.includes( NOFOLLOW_REL ),
+			toData: ( value, { opensInNewTab: opensInNewTabValue } ) => {
```

Comment:
So you updated the `toData` method to pass the whole `linkValue` object so that the other props of the data tree can be accessed?

Reply:
But there could be a bug because (depending on the order in which you add the properties to the mapping) the current value of `opensInNewTab` could be stale and not represent the most recent value. 

I think perhaps we need to be able to access _both_ the current data object _and_ the value coming from `LinkControl`'s `onChange` handler. That way we could access the property from the link value.

Reply:
Actually...maybe I"m confusing myself here. The 2nd argument is _already_ the link value 🤦

Reply:
`nofollow` and `opensInNewTab` both modify `rel` attribute, I am starting to see the `toData` as a reducer here 😳